invertedai/large/initialize.py

```diff
@@ -0,0 +1,391 @@
+import numpy as np
+from random import choices
+
+import invertedai as iai
+from invertedai.large.common import Region
+from invertedai.api.initialize import InitializeResponse
+from invertedai.common import TrafficLightStatesDict, Point
+from invertedai.utils import get_default_agent_attributes
+from invertedai.error import InvertedAIError
+
+AGENT_SCOPE_FOV_BUFFER = 20
+ATTEMPT_PER_NUM_REGIONS = 15
+
+@validate_call
+def get_regions_default(
+    location: str,
+    total_num_agents: Optional[int] = None,
+    area_size: Optional[Tuple[float,float]] = None,
+    map_center: Optional[Tuple[float,float]] = (0.0,0.0),
+    random_seed: Optional[int] = None, 
+    display_progress_bar: Optional[bool] = True
+) -> List[Region]:
+    """
+    A utility function to create a set of Regions to be passed into :func:`large_initialize` in
+    a single convenient entry point. 
+
+    Arguments
+    ----------
+    location:
+        Location name in IAI format.
+
+    total_num_agents:
+        The total number of agents to initialize across all regions.
+
+    area_size:
+        Contains the [width, height] of the rectangular area to be broken into smaller regions.
+
+    map_center:
+        The coordinates of the center of the rectangular area to be broken into smaller regions.
+
+    random_seed:
+        Controls the stochastic aspects of assigning agents to regions for reproducibility.
+
+    display_progress_bar:
+        A flag to control whether a command line progress bar is displayed for convenience.
+    
+    """
+    if area_size is None:
+        location_response = iai.location_info(
+            location = location,
+            rendering_center = map_center
+        )
+        height, width = 100, 100
+        polygon_x, polygon_y = [], []
+        for pt in location_response.bounding_polygon:
+            polygon_x.append(pt.x)
+            polygon_y.append(pt.y)
+        width = max(polygon_x) - min(polygon_x)
+        height = max(polygon_y) - min(polygon_y)
+
+        area_size = tuple([width,height])
+
+    regions = iai.get_regions_in_grid(
+        width = area_size[0], 
+        height = area_size[1],
+        map_center = map_center
+    )
+
+    if total_num_agents is None:
+        total_num_agents = 10*len(regions)
+
+    new_regions = iai.get_number_of_agents_per_region_by_drivable_area(
+        location = location,
+        regions = regions,
+        total_num_agents = total_num_agents,
+        random_seed = random_seed,
+        display_progress_bar = display_progress_bar
+    )
+
+    return new_regions
+
+@validate_call
+def get_regions_in_grid(
+    width: float,
+    height: float,
+    map_center: Optional[Tuple[float,float]] = (0.0,0.0), 
+    stride: Optional[float] = 50.0
+) -> List[Region]:
+    """
+    A utility function to help initialize an area larger than 100x100m. This function breaks up an 
+    area into a grid of 100x100m regions, each of which can be initialized directly.
```

Comment:
Clarify whether the returned regions have any agents.

Reply:
Done.